projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js

```diff
@@ -40,6 +43,30 @@
 				}
 			} );
 		}
+
+		if ( adminMenu ) {
+			var collapseButton = adminMenu.querySelector( '#collapse-button' );
+			// Nav-Unification feature:
+			// Saves the sidebar state in server when "Collapse menu" is clicked.
+			// This is needed so that we update WPCOM for this preference in real-time.
+			if ( collapseButton ) {
+				collapseButton.addEventListener( 'click', function ( event ) {
+					// Let's the core event listener be triggered first.
+					setTimeout( function () {
+						saveSidebarIsExpanded( event.target.parentNode.ariaExpanded );
+					}, 50 );
+				} );
+			}
+		}
+	}
+
+	function saveSidebarIsExpanded( expanded ) {
+		var xhr = new XMLHttpRequest();
+		xhr.open( 'POST', ajaxurl, true );
+		xhr.setRequestHeader( 'X-Requested-With', 'XMLHttpRequest' );
+		xhr.setRequestHeader( 'Content-Type', 'application/x-www-form-urlencoded; charset=UTF-8' );
+		xhr.withCredentials = true;
+		xhr.send( 'action=sidebar_state&expanded=' + expanded );
```

Comment:
You can also use the `wp.ajax` global here to simplify the code: https://github.com/WordPress/WordPress/blob/e445660ee3c4aa204a152e0a6dbd250d1896abde/wp-includes/js/wp-util.js#L44-L51

```js
wp.ajax.xhr.post( 'sidebar_state', { expanded: expanded } );
```

Reply:
disregard this (see https://github.com/Automattic/jetpack/pull/19520#discussion_r613860390).